Installation issue: rdma-core

### Steps to reproduce the issue

```console
$ spack install rdma-core
```


### Error message

This is the end of the spack-build-out.txt:

```
-- Installing: /home/mdorier/daos-test/spack/opt/spack/linux-sandybridge/rdma-core-57.0-i4c6eay2ozicfdbtovh42m3ek6pxaafz/share/man/man8/dump_lfts.8
-- Installing: /home/mdorier/daos-test/spack/opt/spack/linux-sandybridge/rdma-core-57.0-i4c6eay2ozicfdbtovh42m3ek6pxaafz/share/man/man8/dump_mfts.8
CMake Error at pyverbs/cmake_install.cmake:55 (file):
  file cannot create directory: /usr/lib/python3.8/site-packages/pyverbs.
  Maybe need administrative privileges.
Call Stack (most recent call first):
  cmake_install.cmake:175 (include)


make: *** [Makefile:89: install] Error 1
```

It looks like it's trying to install something in a system directory.

Note that I don't see this package depend on Python. I would expect it to depend on it if `+pyverbs` is specified (and this variant is True by default). So my intuition is that if it's being installed in an environment that doesn't have python, and if it can find a python in system directories, it will build pyverbs and try to install it in there.


### Information on your system

* **Spack:** 1.0.0.dev0 (cd3da0e6b7f4bc805777f5fb334e991e45a12262)
* **Builtin repo:** 413ca0539bdb27d73b306f713029c6ac569c9d0f
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-sandybridge

### Additional information

_No response_

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/repos/spack_repo/builtin/packages/rdma_core/package.py
+++ b/repos/spack_repo/builtin/packages/rdma_core/package.py
@@ -87,7 +87,12 @@ class RdmaCore(CMakePackage):
     variant("pyverbs", default=True, description="Build with support for pyverbs")
     variant("man_pages", default=True, description="Build with support for man pages")
 
-    depends_on("c", type="build")  # generated
+    depends_on("c", type="build")
+
+    with when("+pyverbs"):
+        extends("python")
+        depends_on("python", type="build")
+        depends_on("py-cython", type="build")
 
     depends_on("pkgconfig", type="build")
     depends_on("py-docutils", when="+man_pages", type="build")
@@ -117,8 +122,11 @@ class RdmaCore(CMakePackage):
 
         cmake_args.append(self.define_from_variant("ENABLE_STATIC", "static"))
 
-        if self.spec.satisfies("~pyverbs"):
+        if self.spec.satisfies("+pyverbs"):
+            cmake_args.append(self.define("CMAKE_INSTALL_PYTHON_ARCH_LIB", python_platlib))
+        else:
             cmake_args.append("-DNO_PYVERBS=1")
+
         if self.spec.satisfies("~man_pages"):
             cmake_args.append("-DNO_MAN_PAGES=1")
 

--- a/repos/spack_repo/builtin/packages/rdma_core/package.py
+++ b/repos/spack_repo/builtin/packages/rdma_core/package.py
@@ -79,6 +79,16 @@ class RdmaCore(CMakePackage):
 
     patch("libdrm.patch", when="@34:52")
 
+    def patch(self):
+        if self.spec.satisfies("+pyverbs"):
+            # avoid exceeding shebang length limit during build (in particular in CI)
+            filter_file(
+                r"#!${PYTHON_EXECUTABLE}",
+                "#!/usr/bin/env python",
+                "buildlib/Findcython.cmake",
+                string=True,
+            )
+
     variant(
         "static",
         default=True,

